File: TCDNew-master/src/ui.wince/Cen.Wms.Client/Models/Dtos/PurchaseTaskUpdateDto.cs
using System.Collections.Generic;

namespace Cen.Wms.Client.Models.Dtos
{
    public class PurchaseTaskUpdateDto
    {
        public string PurchaseTaskId { get; set; }
        public long PurchaseTaskVersion { get; set; }
        public List<PurchaseTaskLineDto> LinesUpdated { get; set; }
        public List<PurchaseTaskPalletDto> PalletsUpdated { get; set; }
    }
}

File: TCDNew-master/src/ui.android/lib/models/dtos/PurchaseTaskUpdateDto.dart
import 'package:tiq/models/dtos/PurchaseTaskLineDto.dart';
import 'package:tiq/models/dtos/PurchaseTaskPalletDto.dart';

class PurchaseTaskUpdateDto {
  String purchaseTaskId;
  int purchaseTaskVersion;
  List<PurchaseTaskLineDto> linesUpdated;
  List<PurchaseTaskPalletDto> palletsUpdated;

  PurchaseTaskUpdateDto(this.purchaseTaskId, this.purchaseTaskVersion, this.linesUpdated, this.palletsUpdated);

  factory PurchaseTaskUpdateDto.fromJson(Map<String, dynamic> json) {
    PurchaseTaskUpdateDto data;

    if (json != null) {
      final linesUpdated = List<PurchaseTaskLineDto>();
      json['linesUpdated'].forEach((value) => linesUpdated.add(
          PurchaseTaskLineDto.fromJson(value)
      ));

      final palletsUpdated = List<PurchaseTaskPalletDto>();
      json['palletsUpdated'].forEach((value) => palletsUpdated.add(
          PurchaseTaskPalletDto.fromJson(value)
      ));

      data = PurchaseTaskUpdateDto(
          json['purchaseTaskId'].toString(),
          json['purchaseTaskVersion'],
          linesUpdated,
          palletsUpdated
      );
    }

    return data;
  }
}